Modules/_ssl.c

```diff
@@ -2330,6 +2360,170 @@ PySSL_select(PySocketSockObject *s, int writing, _PyTime_t timeout)
     return rc == 0 ? SOCKET_HAS_TIMED_OUT : SOCKET_OPERATION_OK;
 }
 
+/*[clinic input]
+_ssl._SSLSocket.uses_ktls_for_send
+
+Check if the Kernel TLS data-path is used for sending.
+[clinic start generated code]*/
+
+static PyObject *
+_ssl__SSLSocket_uses_ktls_for_send_impl(PySSLSocket *self)
+/*[clinic end generated code: output=f9d95fbefceb5068 input=604d98b67c65e8a7]*/
+{
+#ifdef BIO_get_ktls_send
+    int uses = BIO_get_ktls_send(SSL_get_wbio(self->ssl));
+    return PyBool_FromLong((long)uses);
+#else
+    return Py_False;
```

Comment:
Py_INCREF before returning the singleton anywhere you return Py_False or Py_True.

Reply:
Thanks for the feedback! It looks like `Py_INCREF` for boolean objects has not been needed since this commit https://github.com/python/cpython/commit/ea2c0016502472aa8baa3149050ada776d17a009#diff-cb015dd8a8161c3ca18f71fc78aac892d865a2a096a13ffc03a9cda8b02970ae. [Documentation](https://docs.python.org/3.12/c-api/bool.html#c.Py_False) no longer mentions the need too.

Reply:
Instead use `Py_RETURN_FALSE`:

```suggestion
    Py_RETURN_FALSE;
```